client/src/components/TinymceInlineToolbar/tests/TinymceInlineToolbar-test.js

```diff
@@ -0,0 +1,102 @@
+/* global jest, describe, beforeEach, it, expect, document */
+import { shouldShowToolbar, setupTinyMceInlineToolbar } from '../TinymceInlineToolbar';
+
+jest.unmock('react');
+jest.unmock('react-addons-test-utils');
+
+// Mock tinycme.activeEditor
+class Editor {
+  constructor() {
+    this.callbacks = [];
+    this.contentDocument = document.createDocumentFragment();
+    this.selection = {
+      isCollapsed: () => true,
+    };
+  }
+
+  getContainer() {
+    // Not needed for testing
+  }
+
+  on(event, callback) {
+    this.callbacks[event] = callback;
+  }
+
+  trigger(event, eventData) {
+    const callback = this.callbacks[event];
+    if (typeof callback === 'function') {
+      callback(eventData);
+    }
+  }
+}
+
+describe('shouldShowToolbar function', () => {
+  const tagsToCheck = ['A', 'P', 'EM', 'IMG'];
+
+  it('should return true when one of the tag types is selected and the editor is focused', () => {
+    expect(shouldShowToolbar({ tagName: 'A' }, true, tagsToCheck)).toBe(true);
+  });
+
+  it('should be case-insensitive', () => {
+    expect(shouldShowToolbar({ tagName: 'img' }, true, tagsToCheck)).toBe(true);
+  });
+
+  it('should return false when there isn\'t a seleciton', () => {
+    expect(shouldShowToolbar(null, true, tagsToCheck)).toBe(false);
+  });
+
+  it('should return false when the selected tag is\'t one of the tag types specified', () => {
+    expect(shouldShowToolbar({ tagName: 'STRONG' }, true, tagsToCheck)).toBe(false);
+    expect(shouldShowToolbar({ tagName: null }, true, tagsToCheck)).toBe(false);
+    expect(shouldShowToolbar({ }, true, tagsToCheck)).toBe(false);
+  });
+
+  it('should return false when the editor is not focused', () => {
+    expect(shouldShowToolbar({ tagName: 'IMG' }, false, tagsToCheck)).toBe(false);
+  });
+
+  it('should return false when the specified tag types to check is not an array', () => {
+    expect(shouldShowToolbar({ tagName: 'IMG' }, true, null)).toBe(false);
+    expect(shouldShowToolbar({ tagName: 'IMG' }, true, 3)).toBe(false);
+    expect(shouldShowToolbar({ tagName: 'IMG' }, true, NaN)).toBe(false);
+    expect(shouldShowToolbar({ tagName: 'IMG' }, true, undefined)).toBe(false);
+  });
+});
+
+describe('setupTinyMceInlineToolbar function', () => {
+  global.tinymce = {
+    ui: {
+      Factory: {
+        create: function create() {
+          return {
+            hide: jest.genMockFunction(),
+            remove: jest.genMockFunction(),
+            show: jest.genMockFunction(),
+            renderTo: jest.genMockFunction(),
+            getEl: () => document.createDocumentFragment(),
+          };
+        },
+      },
+    },
+  };
+
+  const editor = new Editor();
```

Comment:
Put `global.tinymce` and `editor` inside a `beforeEach(() => {` block, that way we can ignore mutations and possible side-effects by rebuilding this objects after each `it()` test.

We could mock `editor` with an object that contains the methods we expect to be used, e.g. `on = jest.fn()` and `hide = jest.fn(() => editor)`.